packages/fx-core/src/plugins/resource/bot/configs/localDebugConfig.ts

```diff
@@ -38,6 +38,10 @@ export class LocalDebugConfig {
     this.localRedirectUri = context.localSettings?.bot?.get(
       LocalSettingsBotKeys.BotRedirectUri
     ) as string;
+
+    // To respect existing Bot AAD.
+    this.localBotId = context.envInfo.config.bot?.appId ?? this.localBotId;
```

Comment:
What's the relationship between L43 and L31 to get the value of localBotId? According to the code, L43 will always override the value of localBotId?